File: trunk/Plug-Ins/Interfaces/Interfaces/PluginDetails.vb
Public Class PluginDetails

    Dim strName As String
    Dim strDescription As String
    Dim strAuthor As String
    Dim intVersionMajor As Integer
    Dim intVersionMinor As Integer
    Dim bolhasSettings As Boolean
    Public Sub New(ByVal Name As String, ByVal Description As String, ByVal Author As String, ByVal VersionMajor As Integer, ByVal VersionMinor As Integer, ByVal HasSettings As Boolean)
        strName = Name
        strDescription = Description
        strAuthor = Author
        intVersionMajor = VersionMajor
        intVersionMinor = VersionMinor
        bolhasSettings = HasSettings
    End Sub

    Public ReadOnly Property Name() As String
        Get
            Return strName
        End Get
    End Property

    Public ReadOnly Property Description() As String
        Get
            Return strDescription
        End Get
    End Property

    Public ReadOnly Property Author() As String
        Get
            Return strAuthor
        End Get
    End Property

    Public ReadOnly Property VersionMajor() As String
        Get
            Return intVersionMajor
        End Get
    End Property

    Public ReadOnly Property VersionMinor() As String
        Get
            Return intVersionMinor
        End Get
    End Property
    Public ReadOnly Property HasSettings() As Boolean
        Get
            Return bolhasSettings
        End Get
    End Property
End Class

File: branches/XBMCSkinEditor_2008/XBMCSkinEditor/Components/Interfaces/PluginDetails.cs

namespace SkinEditor.Interfaces
{
    public class PluginDetails
    {

        string strName;
        string strDescription;
        string strAuthor;
        int intVersionMajor;
        int intVersionMinor;
        bool bolhasSettings;

        public PluginDetails(string Name, string Description, string Author, int VersionMajor, int VersionMinor, bool HasSettings)
        {
            strName = Name;
            strDescription = Description;
            strAuthor = Author;
            intVersionMajor = VersionMajor;
            intVersionMinor = VersionMinor;
            bolhasSettings = HasSettings;
        }

        public string Name
        {
            get { return strName; }
        }

        public string Description
        {
            get { return strDescription; }
        }

        public string Author
        {
            get { return strAuthor; }
        }

        public int VersionMajor
        {
            get { return intVersionMajor; }
        }

        public int VersionMinor
        {
            get { return intVersionMinor; }
        }
        public bool HasSettings
        {
            get { return bolhasSettings; }
        }
    }
}
